pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -247,45 +253,48 @@ func validateMachine(machine machinev1.Machine, providerSpec v1beta1.GCPMachineP
 
 // Returns true if machine exists.
 func (r *Reconciler) exists() (bool, error) {
-	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
-		return false, fmt.Errorf("failed validating machine provider spec: %v", err)
-	}
-	zone := r.providerSpec.Zone
-	// Need to verify that our project/zone exists before checking machine, as
-	// invalid project/zone produces same 404 error as no machine.
-	if err := r.validateZone(); err != nil {
-		return false, fmt.Errorf("unable to verify project/zone exists: %v/%v; err: %v", r.projectID, zone, err)
-	}
-	_, err := r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
-	if err == nil {
-		klog.Infof("Machine %q already exists", r.machine.Name)
-		return true, nil
-	}
-	if isNotFoundError(err) {
-		klog.Infof("Machine %q does not exist", r.machine.Name)
-		return false, nil
+	instance, err := r.instanceGet()
+	if err != nil {
+		return false, err
 	}
-	return false, fmt.Errorf("error getting running instances: %v", err)
+	return instance != nil && !haveCreateOperationInProgress(r.machine), nil
```

Comment:
We can save api calls to the cloud provider by testing haveCreateOperationInProgress first.

```go
if haveCreateOperationInProgress(r.machine) {
  return false, nil
}
<get instance, check error>
return instance != nil
```

Reply:
Actually, this would save 2 api calls per loop if we're not finished creating.

Reply:
As things stand today I wanted to the double check, even if it is more costly.

If the annotation is left on the machine (because scope.Close() failed) then this would always return true - which is only one half of the story.